index.bs

```diff
@@ -485,6 +520,155 @@ spec:csp3; type:grammar; text:base64-value
+  1. Let |target| be |global|'s <a>associated Document</a>.
+  1. Let |document URL| be the result of <a>strip URL for use in reports</a> on |target|'s <a>document URL</a>.
+  1. Let |blocked URL| be the result of <a>strip URL for use in reports</a> on |request|'s <a for=request>URL</a>.
+  1. If |block|, <a for=list>for each</a> |endpoint| in |policy|'s <a>endpoints</a>:
+     1. Let |body| be a new {{IntegrityPolicyViolationReportBody}}, initialized as follows:
+                 :   {{IntegrityPolicyViolationReportBody/documentURL}}
+                 ::  |document URL|
+                 :   {{IntegrityPolicyViolationReportBody/blockedURL}}
+                 ::  |blocked URL|
+                 :   {{IntegrityPolicyViolationReportBody/destination}}
+                 ::  |request|'s <a for=request>destination</a>
+                 :   {{IntegrityPolicyViolationReportBody/reportOnly}}
+                 ::  false
+     2. <a>Generate and queue a report</a> with the following arguments:
+              :   <var ignore>context</var>
+              ::  |settings object|
+              :   <var ignore>type</var>
+              ::  "integrity-policy-violation"
+              :   <var ignore>destination</var>
+              ::  |endpoint|
+              :   <var ignore>data</var>
+              ::  |body|
+  1. If |report block|, <a for=list>for each</a> |endpoint| in |policy|'s <a>endpoints</a>:
+     1. Let |report body| be a new {{IntegrityPolicyViolationReportBody}}, initialized as follows:
+                 :   {{IntegrityPolicyViolationReportBody/documentURL}}
+                 ::  |document URL|
+                 :   {{IntegrityPolicyViolationReportBody/blockedURL}}
+                 ::  |blocked URL|
+                 :   {{IntegrityPolicyViolationReportBody/destination}}
+                 ::  |request|'s <a for=request>destination</a>
+                 :   {{IntegrityPolicyViolationReportBody/reportOnly}}
+                 ::  true 
+     2. <a>Generate and queue a report</a> with the following arguments:
+              :   <var ignore>context</var>
+              ::  |settings object|
+              :   <var ignore>type</var>
+              ::  "integrity-policy-violation"
+              :   <var ignore>destination</var>
+              ::  |endpoint|
+              :   <var ignore>data</var>
+              ::  |report body|
+
+  ### Integration ### {#integration}
+
+  A <a>policy container</a> has extra items:
+
+  * <dfn>integrity policy</dfn>, an <a>integrity policy struct</a>.
+  * <dfn>report only integrity policy</dfn>, an <a>integrity policy struct</a>.
+
+  Add an extra step to <a>create a policy container from a fetch response</a> before it returns, that runs
+  <a>parse Integrity-Policy headers</a> with <var ignore>response</var> and <var ignore>result</var>.
+
+  Note: The creation of a policy container only happens on navigation responses, but there's no
+        current way to assert that as part of the algorithm.
```

Comment:
What? Policy containers are also used by workers and worklets.

Reply:
Oops, I totally missed the call to creating a policy container from a fetch response from https://html.spec.whatwg.org/#initialize-worker-policy-container. I'll remove this note.

Reply:
Removed